src/libgit2/sparse.c

```diff
@@ -415,7 +415,7 @@ int git_sparse_checkout__reapply(git_repository *repo, git_sparse *sparse)
 		/* Don't touch files that aren't current */
 		if ((error = git_status_file(&status_flags, repo, entry->path)) < 0)
 			goto done;
-		if (status_flags != GIT_STATUS_CURRENT)
+		if (status_flags != GIT_STATUS_CURRENT && status_flags != GIT_STATUS_WT_DELETED)
```

Comment:
status_flags are a bitmask, so seeing equality operators is a bit strange.

Stepping back – GIT_STATUS_CURRENT are files where the Index and working directory match the HEAD ref. git reapply doesn't want obliterate an unstaged or staged change. Arguably this should be allowed with a force flag, regardless, the underlying problem is that git_status_file is showing GIT_STATUS_WT_DELETED when a file is missing from the worktree (I'm guessing because it is not in the sparse checkout).

Perhaps the root fix it to modify git_status_file to that it doesn't return GIT_STATUS_WT_DELETE for sparse-no-checkout files?